format-config-v1.go

```diff
@@ -37,66 +43,191 @@ type formatConfigV1 struct {
 	XL      *xlFormat `json:"xl,omitempty"`
 }
 
-// FIXME: currently we don't check single exportPath which uses FS layer.
+// validateXLJBOD - validate xl jbod order if they are consistent.
+func validateXLJBOD(jbod []string, jsonMap map[int]formatConfigV1) error {
+	jbodStr := strings.Join(jbod, ", ")
+	for _, format := range jsonMap {
+		if format.XL == nil {
+			continue
+		}
+		savedJBODStr := strings.Join(format.XL.JBOD, ", ")
+		if jbodStr != savedJBODStr {
+			return errors.New("Inconsistent disks.")
+		}
+	}
+	return nil
+}
 
-// loadFormatXL - load XL format.json.
-func loadFormatXL(storage StorageAPI) (xl *xlFormat, err error) {
-	offset := int64(0)
-	r, err := storage.ReadFile(minioMetaBucket, formatConfigFile, offset)
-	if err != nil {
-		return nil, err
+// arrangeDiskInOrder - arrange disks in saved order.
+func arrangeDiskInOrder(storageDisks []StorageAPI, jbodMap map[int]formatConfigV1) ([]StorageAPI, error) {
+	var savedJBOD []string
+	for index := range storageDisks {
+		if format, ok := jbodMap[index]; ok {
```

Comment:
Change this logic to simply deepEquals first JBOD to the rest. If any fails, the whole check fails.